src/segment-loader.js

```diff
@@ -542,7 +542,11 @@ export default class SegmentLoader extends videojs.EventTarget {
     // decrease in network performance or a server issue.
 
     // don't timeout if we are on the last un-blacklisted playlist
-    segmentInfo.dontTimeout ? requestTimeout = 0 : requestTimeout = (segment.duration * 1.5) * 1000;
+    if (segmentInfo.dontTimeout) {
+      requestTimeout = 0;
+    } else {
+      requestTimeout = (segment.duration * 1.5) * 1000;
+    }
```

Comment:
This is not at all the solution was discussed at kickoff.

See the `this.hls_.xhr` function being invoked a few lines below this one? The first parameter is an object. The idea was that the base of that object would come from the MasterPlaylistController and have things like `timeout` and `withCredentials` set *there*. Then that object get passed along to the SegmentLoaders whenever a playlist update happened so that it was always current. In SegmentLoader, the default xhr options from MasterPlaylistController would get merged with an object with the `uri` and `responseType` properties whenever we need to call `xhr` to create the final xhr-related options that we will pass along to the `xhr` function.